### PR TITLE
RHOAIENG-3943 Update Model Registry documentation

### PR DESCRIPTION
**Pull request by tarilabs**
## Description
I followed same style per https://issues.redhat.com/browse/RHOAIENG-3946

I've used as sources:
- "Model Registry Architecture" 
- "Model Registry Proposal" 
- https://issues.redhat.com/browse/RHOAIENG-1071 
- https://github.com/opendatahub-io/model-registry-bf4-kf/issues/249 
- https://github.com/opendatahub-io/odh-model-controller/pull/135

Since "Model Registry Tenancy Proposal" is still under finalization I didn't include it yet in the current PR, we can expand on it later once the ADR is approved.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: architecture/components/model-registry/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+We envision this to be more of a backing store for various stages of MLOps that can log user flow of a model development and deployment. The model registry will meet a data scientist's need to be able to visualize a model’s lineage and trace back the training executions, parameters, metrics, etc. It will also help deployment engineers visualize model pipeline events, actions, progress through deployment stages, etc. 
+
+## Goals 
+- Associate metadata from as training, experimentation, studies and their metrics, with a model
+- Build a catalog of models and manage model versions for deployment
+- Management of model to different deployment environments
+- Build an open-source solution with vibrant community (Kubeflow)
+- Build a Kube Native solution
+- Avoid reinvention unless it is mandatory
```
Comment (etirelli) [inline architecture/components/model-registry/README.md:26]: since this is the architecture documentation, I would remove lines 24 and 26 above as they are more relevant for community and development

File: architecture/components/model-registry/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+## Architecture
+
+Google community project [ML-Metadata](https://github.com/google/ml-metadata) will be used as the core component to build the Model Registry. ML-Metadata provides a very extensible schema that is generic, similar to a key-value store, but also allows for the creation of logical schemas that can be queried as if they were physical schemas. Those can be manipulated using their bindings in the Python library. We intend to use this model to extend and provide metadata storage services for model serving, also known as Model Registry.
```
Comment (etirelli) [inline architecture/components/model-registry/README.md:30]: s/"will be used"/"is used"? 
s/"we intend to use"/"we use"?

**Review by etirelli — CHANGES_REQUESTED** [review]
made some minor editorial comments, but I suggest rewording the whole document to reflect what is the model server component (present tense), instead of what is intended for the future.

File: architecture/components/model-registry/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+## Model Registry High Level Architecture 
+![Model Registry High Level Architecture](./model-registry-overview.jpg)
+
+> [!NOTE]  
+> The Model Registry is a passive repository for metadata and is not meant to be a Control Plane. It does not perform any orchestration or expose APIs to perform actions on underlying OpenShift AI components. 
+
+We envision this to be more of a backing store for various stages of MLOps that can log user flow of a model development and deployment. The model registry will meet a data scientist's need to be able to visualize a model’s lineage and trace back the training executions, parameters, metrics, etc. It will also help deployment engineers visualize model pipeline events, actions, progress through deployment stages, etc.
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply architecture/components/model-registry/README.md:18]:
```suggestion
The model registry is a backing store for various stages of MLOps that can log user flow of a model development and deployment. The model registry meets a data scientist's need to be able to visualize a model’s lineage and trace back the training executions, parameters, metrics, etc. It also help deployment engineers visualize model pipeline events, actions, progress through deployment stages, etc. 
```

File: architecture/components/model-registry/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+## Model Registry High Level Architecture 
+![Model Registry High Level Architecture](./model-registry-overview.jpg)
+
+> [!NOTE]  
+> The Model Registry is a passive repository for metadata and is not meant to be a Control Plane. It does not perform any orchestration or expose APIs to perform actions on underlying OpenShift AI components. 
+
+We envision this to be more of a backing store for various stages of MLOps that can log user flow of a model development and deployment. The model registry will meet a data scientist's need to be able to visualize a model’s lineage and trace back the training executions, parameters, metrics, etc. It will also help deployment engineers visualize model pipeline events, actions, progress through deployment stages, etc. 
+
+## Goals 
+- Associate metadata from as training, experimentation, studies and their metrics, with a model
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply architecture/components/model-registry/README.md:21]:
```suggestion
- Associate metadata from training, experimentation, studies and their metrics, with a model
```

File: architecture/components/model-registry/README.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+- Build a catalog of models and manage model versions for deployment
+- Management of model to different deployment environments
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply architecture/components/model-registry/README.md:23]:
```suggestion
- Management of model for multiple deployment environments
```

File: architecture/components/model-registry/README.md (hunk 1 above)
Comment (tarilabs) [reply architecture/components/model-registry/README.md]: done with https://github.com/opendatahub-io/architecture-decision-records/pull/34/commits/2662f268e63dd739be10889d1e4b1c7295587db6

File: architecture/components/model-registry/README.md (hunk 2 above)
Comment (tarilabs) [reply architecture/components/model-registry/README.md:28]:
```suggestion
Google community project [ML-Metadata](https://github.com/google/ml-metadata) is used as the core component to build the Model Registry. ML-Metadata provides a very extensible schema that is generic, similar to a key-value store, but also allows for the creation of logical schemas that can be queried as if they were physical schemas. Those can be manipulated using their bindings in the Python library. We use this model to extend and provide metadata storage services for model serving, also known as Model Registry.
```

**Comment by tarilabs** [comment]
> made some minor editorial comments, but I suggest rewording the whole document to reflect what is the model server component (present tense), instead of what is intended for the future.

thank you for the feedback @etirelli , I believe I rewritten all future tense sentences I could find to present tense with https://github.com/opendatahub-io/architecture-decision-records/pull/34/commits/4dd3c077b0813126372aa2044da47b7931cd6f31

**Review by etirelli — APPROVED** [review]
